smqtk_classifier/interfaces/classify_image.py

```diff
@@ -0,0 +1,145 @@
+import abc
+import logging
+from typing import Generator, Hashable, Iterable, Iterator, Sequence, Union
+
+import numpy as np
+from PIL import Image
+
+from smqtk_core import Plugfigurable
+
+from smqtk_classifier.interfaces.classification_element import (
+    CLASSIFICATION_DICT_T
+)
+
+
+IMAGE_ITER_T = Union[Image.Image, Iterable[Image.Image]]
+LOG = logging.getLogger(__name__)
+
+
+class ClassifyImage (Plugfigurable):
+    """
+    Interface for algorithms that classify input images into discrete
+    labels and/or label confidences.
+    """
+
+    @abc.abstractmethod
+    def get_labels(self) -> Sequence[Hashable]:
+        """
+        Get the sequence of class labels that this classifier can classify
+        images into. This includes the negative or background label if the
+        classifier embodies such a concept.
+
+        :return: Sequence of possible classifier labels.
+        :rtype: collections.abc.Sequence[collections.abc.Hashable]
+
+        :raises RuntimeError: No model loaded.
+
+        """
+
+    @abc.abstractmethod
+    def _classify_arrays(self, img_iter: IMAGE_ITER_T) -> Iterator[CLASSIFICATION_DICT_T]:
+        """
+        Overridable method for classifying an iterable of PIL.Image.Image
+        whose vectors should be classified.
+
+        At this level, all input arrays are guaranteed to be of consistent
+        dimensionality.
+
+        *Remember:* A single-pass `Iterator` is a valid `Iterable`. If an
+        implementation needs to pass over the input multiple times, either
+        ensure you are receiving an ndarray, or *not* and Iterator.
+
+        Each classification mapping should contain confidence values for each
+        label the configured model contains.
+        Implementations may act in a discrete manner whereby only one label is
+        marked with a ``1`` value (others being ``0``), or in a continuous
+        manner whereby each label is given a confidence-like value in the
+        [0, 1] range.
+
+        :param img_iter:
+            Iterable of arrays to be classified.
+
+        :return: Iterator of dictionaries, parallel in association to the input
+            images. Each dictionary should map labels to associated
+            confidence values.
+        """
+
+    @staticmethod
+    def _assert_array_dim_consistency(img_iter: IMAGE_ITER_T) -> IMAGE_ITER_T:
```

Comment:
This method does not seem to be used. Did you happen to mean to use this in the `classify_arrays` method, or is this a copy-paste left-over?

Reply:
I assume a copy-paste left-over since some of the content here does not make sense for the `PIL.Image.Image` type (e.g. `img_iter.dtype`).